src/session.js

```diff
@@ -47,9 +52,17 @@ function tryReadingIframeLocation(iframe) {
             // we can sort of detect this by checking the location of the iframe
             // if the contentDocument object cannot be accessed, its due to a security error.
             // security error will occur when the frame is on a different origin
-            resolve(iframe.contentDocument.location);
+
+            // Heuristic to detect google auth error in the iframe document
+            var textContent = iframe.contentDocument.body.textContent;
```

Comment:
If the login page is on `media-auth.gutools` and this script is running on media.gutools, will we still have access to the `iframe.contentDocument`?